mule-packager/src/main/java/org/mule/tools/api/validation/DependencyValidator.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Mule ESB Maven Tools
+ * <p>
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * <p>
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.tools.api.validation;
+
+import org.mule.maven.client.api.model.BundleDependency;
+import org.mule.tools.api.classloader.model.ArtifactCoordinates;
+import org.mule.tools.api.exception.ValidationException;
+
+import java.util.List;
+
+/**
+ * Should implement logic for validating a {@link BundleDependency}
+ */
+public interface DependencyValidator {
+
+  /**
+   * Validates that the {@link ArtifactCoordinates} is correct.
+   * @param dependency the dependency to be validated
+   * @return true if the dependency is valid, false otherwise
+   * @throws ValidationException if there is a problem with the {@link ArtifactCoordinates}
+   * and validation should be stopped.
+   */
+  boolean isValid(ArtifactCoordinates dependency) throws ValidationException;
```

Comment:
Not sure if we need this method, since the areDependenciesValid will always iterate through a list.